iOS/Flows/Conversation/Cells/MessageCell.swift

```diff
@@ -184,102 +200,45 @@ extension MessageCell: UICollectionViewDataSource, UICollectionViewDelegateFlowL
     }
 }
 
-/// A cell for displaying individual the root message and replies within the MessageCell.
-class MessageSubcell: UICollectionViewCell {
-
-    /// A rounded and colored background view for the message. Changes color based on the sender.
-    let backgroundColorView = UIView()
-    /// Text view for displaying the text of the message.
-    let textView = MessageTextView()
-    /// A label to show the total number of replies for the root message.
-    let replyCountLabel = Label(font: .smallBold, textColor: .lightGray)
-
-    /// Where this cell appears on the z-axis stack of messages. 0 means the item closest to the user.
-    private var stackIndex = 0
-    /// How much to scale the size of the background view.
-    private var scaleFactor: CGFloat {
-        return 1 - CGFloat(self.stackIndex) * 0.05
-    }
-
-    override init(frame: CGRect) {
-        super.init(frame: frame)
-        self.intitializeViews()
-    }
-
-    required init?(coder: NSCoder) {
-        super.init(coder: coder)
-        self.intitializeViews()
-    }
-
-    private func intitializeViews() {
-        self.contentView.addSubview(self.backgroundColorView)
-        self.backgroundColorView.roundCorners()
-
-        self.backgroundColorView.addSubview(self.textView)
-        self.textView.isScrollEnabled = false
-        self.textView.isEditable = false
-        self.textView.textAlignment = .center
-
-        self.backgroundColorView.addSubview(self.replyCountLabel)
-    }
-
-    override func layoutSubviews() {
-        super.layoutSubviews()
+// MARK: - UIContextMenuInteractionDelegate
 
-        self.backgroundColorView.width = self.width * self.scaleFactor
-        self.backgroundColorView.expandToSuperviewHeight()
-        self.backgroundColorView.centerOnXAndY()
+extension MessageCell: UIContextMenuInteractionDelegate {
 
-        self.textView.expandToSuperviewWidth()
-        self.textView.sizeToFit()
-        self.textView.centerOnXAndY()
+    func contextMenuInteraction(_ interaction: UIContextMenuInteraction,
+                                configurationForMenuAtLocation location: CGPoint) -> UIContextMenuConfiguration? {
```

Comment:
Making the message cell a context menu delegate so it can handle creating its own menus.